output/schema.json

```
@@ -19,7 +19,9 @@
         },
         "absoluteKeywordLocation": {
           "type": "string",
-          "format": "uri"
+          "format": "iri",
+          "$comment": "any fragment must be non-empty, and use json-pointer syntax",
+          "pattern": "^[^#]*(#(/([^~]|~[01])*)*)?$"
```

The comment says must be non-empty, but this does allow an empty fragment.

```suggestion
          "pattern": "^[^#]*(#(/([^~/]|~[01])*)+)?$"
```

The change to + requires a non-empty pointer; disallowing the / from the reference-token character range is better consistent with its spec's ABNF (and iriReferenceToSchemaString above), though it doesn't ultimately change what the regex matches.

And of course %-encoding needs fixing here too.

---

actually I think we need to allow empty fragments, so it's the comment that is wrong here.

---

@karenetheridge Agreed, you need to be able to represent the root with a JSON Pointer fragment, which would be an empty fragment.